pkg/controller/job/job_controller.go

```diff
@@ -1422,7 +1450,7 @@ func (jm *Controller) manageJob(ctx context.Context, job *batch.Job, jobCtx *syn
 		}
 	}
 
-	rmAtLeast := active - wantActive
+	rmAtLeast := (active + terminating) - wantActive
```

Comment:
let's suppose this is a regular job. It has :
```
active=10
terminating=3
wantActive=4
```
Since `active` does not include terminating, you need to delete 6 pods. So IIUC this bit should not be changed.